### PR TITLE
[Web] Complete visual redesign — replace default Blazor/Bootstrap styling

### LINKED ISSUE #137: [Web] Complete visual redesign — replace default Blazor/Bootstrap template styling
## Problem

The Blazor Web app is still running on the stock `dotnet new blazor` scaffold visuals: default Bootstrap theme (`wwwroot/lib/bootstrap`), the generic starter `wwwroot/app.css` (plain `Helvetica Neue` stack, no color palette/typography system), the default navy `NavMenu.razor.css` sidebar, and the placeholder favicon. Milestone #18 (App Layout Redesign) reworked the *layout/UX* of Library, Show Detail, and Up Next, but per the design decision in #123:

> Visual style: none of the 4 candidates was picked as-is — going with original styling built to fit the layout decisions below, rather than a predefined aesthetic.

That "original styling" was never actually produced — the layout changes shipped on top of the unmodified default template, so the app still visually reads as a generic Blazor starter rather than a podcast app with its own identity.

## Scope

- Establish an actual visual identity: color palette, typography, spacing/elevation system — replacing the default Bootstrap blue/gray theme and system font stack.
- Apply it consistently across the whole app, not just the milestone #18 screens: Library, Show Detail, Up Next, Subscriptions, Playlists, Search, Settings, New Episodes, auth/sign-in, and the nav sidebar.
- Replace the placeholder favicon/branding assets.
- Decide whether to keep Bootstrap as the underlying CSS framework (retheme it via Sass variables / CSS custom properties) or move away from it — either is reasonable, but the current "untouched default" state isn't.
- Consider dark mode support given the app's audio/podcast context (many competitor apps default to dark).

## Out of scope

- Any further layout/UX restructuring (that's milestone #18's job) — this issue is about visual styling on top of whatever layout exists.
- The iOS app (SwiftUI has its own styling system; track separately if needed).

## Related

- #123 — design decision doc; the "original styling" direction referenced above was never executed.
- Milestone #18 (App Layout Redesign) — layout work this depends on / should follow.

### PR DESCRIPTION
**Pull request by two4suited**
## Summary

Establishes Kuulla's own visual identity in place of the untouched `dotnet new blazor` scaffold styling (default Bootstrap blue/gray theme, system font stack, navy sidebar gradient, placeholder favicon):

- **Design system** — CSS custom properties (`--k-*`) for color, radius, shadow, and font tokens, defined for both light and dark themes and layered over Bootstrap 5.3's component classes so the retheme applies to every page (Library, Show Detail, Up Next, Subscriptions, Playlists, Search, Settings, New Episodes, auth) without per-page edits — only `wwwroot/app.css` and the layout/nav files changed.
- **Palette** — a single coral accent (`#ff5a36`) on near-black surfaces in dark mode (the default) or warm-white surfaces in light mode, replacing Bootstrap's default blue and the scaffold's navy/purple sidebar gradient.
- **Typography** — Manrope (Google Fonts) in place of the default system font stack.
- **Dark mode** — implemented via Bootstrap's built-in `data-bs-theme` attribute; defaults to the OS preference, persisted per-user in `localStorage`, with a toggle in the sidebar (`theme.js`). Handles Blazor's enhanced-navigation DOM patching, which otherwise strips the attribute between page loads.
- **Branding** — new waveform mark used as the nav brand icon and as `favicon.svg`/`favicon.png`, replacing the placeholder favicon.

Kept Bootstrap as the underlying framework and retheme it, rather than replacing it, since its 5.3 variable-driven components made a consistent app-wide retheme achievable without restructuring markup (out of scope per the issue).

An independent review (medium effort, 2 finder angles) caught and fixed: an unthemed episode-status badge in `EpisodeDetail.razor`, a `theme-color` meta tag that didn't track the active theme, a missing focus ring on the new theme-toggle button, and a toggle edge case around OS theme changes.

## Test plan

- [x] `dotnet build Kuulla.sln` — succeeds
- [x] `dotnet test tests/Kuulla.Web.Tests` — 81/81 passing
- [x] Manually verified via Aspire + browser: Library, Search, Show Detail, Episode Detail, Subscriptions, Up Next, Settings in both light and dark theme; theme toggle persists across navigation and page reload

Closes #137

### CODE REVIEW

File: src/Kuulla.Web/wwwroot/theme.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+(function () {
+    var STORAGE_KEY = "kuulla-theme";
+
+    function preferredTheme() {
+        var stored = localStorage.getItem(STORAGE_KEY);
+        if (stored === "light" || stored === "dark") {
+            return stored;
+        }
+        return window.matchMedia("(prefers-color-scheme: light)").matches ? "light" : "dark";
+    }
```
[the remark this answers is not in the document]
Comment (two4suited) [reply src/Kuulla.Web/wwwroot/theme.js]: Fixed — localStorage.getItem/setItem are now wrapped in try/catch (treating a throw as no stored preference / a best-effort no-op write). See theme.js.

File: src/Kuulla.Web/wwwroot/theme.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        toggle: function () {
+            var next = currentTheme() === "dark" ? "light" : "dark";
+            localStorage.setItem(STORAGE_KEY, next);
+            applyTheme(next);
+            return next;
+        },
+    };
+
+    applyTheme(preferredTheme());
+
```
[the remark this answers is not in the document]
Comment (two4suited) [reply src/Kuulla.Web/wwwroot/theme.js:62]: Fixed — added a matchMedia change listener that follows the OS theme live, but only while the user hasn't made an explicit choice (i.e. readStoredTheme() returns null).

File: src/Kuulla.Web/wwwroot/theme.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    function registerEnhancedLoadHandler() {
+        if (window.Blazor && typeof window.Blazor.addEventListener === "function") {
+            window.Blazor.addEventListener("enhancedload", function () {
+                applyTheme(preferredTheme());
+            });
+        } else {
+            // blazor.web.js loads after this script; retry until it's ready.
+            setTimeout(registerEnhancedLoadHandler, 50);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (two4suited) [reply src/Kuulla.Web/wwwroot/theme.js:92]: Fixed — capped retries at 100 (~5s) and give up silently after that instead of polling forever.

File: src/Kuulla.Web/wwwroot/theme.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        return window.matchMedia("(prefers-color-scheme: light)").matches ? "light" : "dark";
+    }
+
+    function applyTheme(theme) {
+        document.documentElement.setAttribute("data-bs-theme", theme);
+        var meta = document.querySelector('meta[name="theme-color"]');
+        if (meta) {
+            meta.setAttribute("content", theme === "dark" ? "#171A21" : "#faf9f7");
+        }
+    }
```
[the remark this answers is not in the document]
Comment (two4suited) [reply src/Kuulla.Web/wwwroot/theme.js:45]: Fixed — applyTheme() now sets aria-pressed on the toggle button on every theme change (initial load, toggle click, OS change, and enhanced-nav re-apply).

File: src/Kuulla.Web/Components/Layout/NavMenu.razor.css [hunk 5]
```diff
@@ -3,52 +3,112 @@
...
     position: relative;
     width: 1.25rem;
     height: 1.25rem;
     margin-right: 0.75rem;
     top: -1px;
-    background-size: cover;
+    background-color: currentColor;
+    -webkit-mask-size: cover;
+    mask-size: cover;
 }
```
[the remark this answers is not in the document]
Comment (two4suited) [reply src/Kuulla.Web/Components/Layout/NavMenu.razor.css:96]: Fixed — added explicit mask-repeat: no-repeat and mask-position: center (with -webkit- prefixes) to the .bi rule.

File: src/Kuulla.Web/Components/Layout/NavMenu.razor [hunk 6]
```diff
@@ -1,6 +1,16 @@
...
-        <a class="navbar-brand" href="">Kuulla.Web</a>
+        <a class="navbar-brand" href="">
+            <svg class="brand-mark" viewBox="0 0 32 32" aria-hidden="true">
+                <rect x="2" y="9" width="4" height="14" rx="2" />
+                <rect x="9" y="5" width="4" height="22" rx="2" />
+                <rect x="16" y="11" width="4" height="10" rx="2" />
+            </svg>
+            Kuulla
+        </a>
+        <button type="button" class="theme-toggle" title="Toggle theme" aria-label="Toggle color theme" onclick="window.kuullaTheme?.toggle()">
```
[the remark this answers is not in the document]
Comment (two4suited) [reply src/Kuulla.Web/Components/Layout/NavMenu.razor]: Fixed — added a static aria-pressed="true" default on the button in NavMenu.razor, matching the dark-first default theme; JS corrects it immediately on load.
